source/backend/api/Areas/Notes/Controllers/NoteController.cs

```diff
@@ -49,18 +57,56 @@ public NoteController(INoteService noteService, IMapper mapper)
         /// Add the specified note.
         /// </summary>
         /// <param name="type">The parent entity type.</param>
-        /// <param name="noteModel">The note to add.</param>
+        /// <param name="model">The note to add.</param>
         /// <returns></returns>
         [HttpPost("{type}")]
         [HasPermission(Permissions.NoteAdd)]
         [Produces("application/json")]
         [ProducesResponseType(typeof(EntityNoteModel), 200)]
         [SwaggerOperation(Tags = new[] { "note" })]
         [TypeFilter(typeof(NullJsonResultFilter))]
-        public IActionResult AddNote(NoteType type, [FromBody] EntityNoteModel noteModel)
+        public IActionResult AddNote(NoteType type, [FromBody] EntityNoteModel model)
         {
-            var createdNote = _noteService.Add(type, noteModel);
-            return new JsonResult(createdNote);
+            _logger.LogInformation(
+                "Request received by Controller: {Controller}, Action: {ControllerAction}, User: {User}, DateTime: {DateTime}",
+                nameof(NoteController),
+                nameof(AddNote),
+                User.GetUsername(),
+                DateTime.Now);
+
+            switch (type)
+            {
+                case NoteType.Acquisition_File:
+                    var acquisitionFileNoteEntity = _mapper.Map<PimsAcquisitionFileNote>(model);
+                    var acquisitionFileNote = _noteService.AddAcquisitionFileNote(acquisitionFileNoteEntity);
+                    return new JsonResult(_mapper.Map<EntityNoteModel>(acquisitionFileNote));
+                case NoteType.Disposition_File:
+                    var dispositionFileNoteEntity = _mapper.Map<PimsDispositionFileNote>(model);
+                    var dispositionFileNote = _noteService.AddDispositionFileNote(dispositionFileNoteEntity);
+                    return new JsonResult(_mapper.Map<EntityNoteModel>(dispositionFileNote));
```

Comment:
This whole logic was being handled by the service (including mapping the model to entities). It is now moved up to the controller who is responsible for data marshalling/conversion. The service now only knows about Pims entities